pymongo/client_session.py

```diff
@@ -393,26 +396,46 @@ def callback(session):
           with client.start_session() as session:
               session.with_transaction(callback)
 
-        In an ideal scenario, `with_transaction` will start a transaction,
-        execute the callback once, and commit the transaction. In the event of
-        error, the commit or entire transaction may need to be retried and
-        thus the callback may be invoked multiple times. Even though the
-        callback may be invoked multiple times, `with_transaction` ensures
+        In the event an exception, `with_transaction` may retry the commit or
+        entire transaction and thus ``callback`` may be invoked multiple times.
+        Applications should take care when writing a ``callback`` that changes
```

Comment:
When writing ``callback``s that modify application state, developers should be mindful of the possibility that a ``callback`` may be executed multiple times by a single call to ``with_transaction``.